app/components/work_packages/activities_tab/journals/item_component.rb

```diff
@@ -0,0 +1,153 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module WorkPackages
+  module ActivitiesTab
+    module Journals
+      class ItemComponent < ApplicationComponent
+        include ApplicationHelper
+        include WorkPackages::ActivitiesTab::SharedHelpers
+        include OpPrimer::ComponentHelpers
+        include OpTurbo::Streamable
+
+        def initialize(journal:, filter:, state: :show)
+          super
+
+          @journal = journal
+          @state = state
+          @filter = filter
+        end
+
+        def content
+          case state
+          when :show
+            render(WorkPackages::ActivitiesTab::Journals::ItemComponent::Show.new(**child_component_params))
+          when :edit
+            render(WorkPackages::ActivitiesTab::Journals::ItemComponent::Edit.new(**child_component_params))
+          end
+        end
```

Comment:
```suggestion
        def content_component
          case state
          when :show
            WorkPackages::ActivitiesTab::Journals::ItemComponent::Show
          when :edit
            WorkPackages::ActivitiesTab::Journals::ItemComponent::Edit
          end
        end
```

instead of rendering in here, why not simply do the class check and render it in the view instead?

Reply:
I put it into the view